homeassistant/components/enphase_envoy/sensor.py

```diff
@@ -115,21 +136,36 @@ def icon(self):
         """Icon to use in the frontend, if any."""
         return ICON
 
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        if self._type == "inverters":
+            return {"last_reported": self._last_reported}
+
+        return None
+
     async def async_update(self):
         """Get the energy production data from the Enphase Envoy."""
-
         if self._type != "inverters":
-            _state = await getattr(EnvoyReader(self._ip_address), self._type)()
+            _state = await getattr(self._envoy_reader, self._type)()
             if isinstance(_state, int):
                 self._state = _state
             else:
                 _LOGGER.error(_state)
                 self._state = None
 
-        elif self._type == "inverters":
-            inverters = await (EnvoyReader(self._ip_address).inverters_production())
+        elif self._type == "inverters" and self._inv_auth:
```

Comment:
Do we need the `_inv_auth` attribute? It seems to always correlate with type `"inverters"`.

Reply:
My thought was that `async_update()` would fail every polling period.  I now see that the inverter entity won't be created during `async_setup_platform()` when an exception occurs.  So no, the `_inv_auth` attribute is not needed.  Awesome!  I'll remove that attribute and check.